src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.Invoke.cs

```diff
@@ -40,10 +40,7 @@ internal static unsafe void InvokeJSFunctionByName<T1>(string functionName, T1 a
         {
             const int argsCount = 1;
             var extraBufferLength = 0;
-            var excSig = GetExceptionMeta();
-            var resSig = GetArgumentMeta<TResult>(ref extraBufferLength);
-            var arg1Sig = GetArgumentMeta<T1>(ref extraBufferLength);
-            var signature = stackalloc JavaScriptMarshallerArgSig[] { excSig, resSig, arg1Sig };
+            var signature = stackalloc JavaScriptMarshallerArgSignature[] { exceptionSig, GetArgumentSignature<TResult>(ref extraBufferLength), GetArgumentSignature<T1>(ref extraBufferLength) };
```

Comment:
the magic extra slots are annoying here too but i'm not sure if it's straightforward to get rid of them